### PR TITLE
Elasticsearch: Create the id field for logs dataframes

### PR DESCRIPTION
**Pull request by gabor**
(fixes half of https://github.com/grafana/grafana/issues/71623)

logs-datasources can provide in the dataframes an `id` column. that is used for certain features, for example they become react-keys on the log-rows.

currently the ElasticSearch datasource does not do this, this PR adds this (only in frontend-mode. there will be a separate PR for backend-mode).

we use the field `_id` from Elastic, which is an unique-id for the given index. to create the fully-unique-id, we combine the index-name and the `_id` field to create the `id`.

how to test:
1. go to explore, show some elastic logs
2. make sure you have the react-devtools
3. inspect a LogRow component. check the react-key on it. verify that it starts with a number, then dash, then index-name, then dash, then some random characters. (on main-branch the react-keys look like : `k_A_499`)

### CODE REVIEW

File: public/app/plugins/datasource/elasticsearch/ElasticResponse.test.ts [hunk 1]
```diff
@@ -1530,6 +1542,104 @@ describe('ElasticResponse', () => {
...
+    const mdf = new MutableDataFrame(df);
+    addIdFieldToDataFrame(mdf);
+
+    const idFields = mdf.fields.filter((f) => f.name === 'id');
+    expect(idFields).toHaveLength(1);
+    expect(idFields[0]?.values).toStrictEqual(['x1', 'x2']);
+  });
+
+  it('does not add id if not logs-request', () => {
+    // FIXME
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/elasticsearch/ElasticResponse.test.ts:1639]: Is this done by 
```
it('should not have an id-field created', () => {
      const idFields = result.data[0].fields.filter((f) => f.name === 'id');
      expect(idFields).toHaveLength(0);
    });
```

or do you still plan to add it?

**Review by ivanahuckova — COMMENTED** [review]
Left a comment bellow, otherwise LGTM

File: public/app/plugins/datasource/elasticsearch/ElasticResponse.test.ts (hunk 1 above)
Comment (gabor) [reply public/app/plugins/datasource/elasticsearch/ElasticResponse.test.ts:1639]: good catch! will check that ...

Comment (gabor) [reply public/app/plugins/datasource/elasticsearch/ElasticResponse.test.ts]: yes, that case is handled by the other test, removed this test.

**Comment by gabor** [comment]
we may approach this problem slightly differently, i switched the PR to draft for now.

**Pull request closed without merging**